ALEC-227: ALEC docs: update installation section (#101)
OpenNMS/alec: docs/modules/install/pages/pre_install.adoc

> = Pre-Installation Tasks
> :imagesdir: ../assets/images

Reviewer: ```suggestion

:imagesdir: ../assets/images
= Pre-Installation Tasks
```

:imagesdir: ../assets/images
= Pre-Installation Tasks

This section describes the configuration tasks you must complete before installing ALEC.

> == OpenNMS configuration

Reviewer: I think we should use the tabbed block here too, if we can. :)

> === Set $OPENNMS_HOME variable

Reviewer: ```suggestion
=== Set `$OPENNMS_HOME` variable
```

The ALEC installation instructions refer to the `$OPENNMS_HOME` environment variable, which must be set.

On RHEL or CentOS, use the following command:

```
export OPENNMS_HOME=/opt/opennms
```

On Ubuntu or Debian, use the following command:

```
export OPENNMS_HOME=/usr/share/opennms
```

> === Validate your OpenNMS deployment

Reviewer: ```suggestion
=== Verify your OpenNMS installation
```

From the OpenNMS Karaf shell, run the following to help validate your deployment:

```
health:check

> ```

Reviewer: ```suggestion
Run the `health:check` command in an OpenNMS Karaf shell to verify the status of your OpenNMS instance.
```

If the health check returns an `everything is awesome` message, continue with the next pre-installation task.

> An `Oh no, something is wrong` message indicates a problem you must address.

Reviewer: ```suggestion
If the health check returns "Everything is awesome," continue to the next pre-installation task.
An "Oh no, something is wrong" error message indicates a problem that you must address before moving forward.
```

The most common result you may see involves a missing Elasticsearch configuration, similar to the following:

```
admin@opennms> health:check
Verifying the health of the container

Verifying installed bundles                    [ Success  ]
Connecting to ElasticSearch ReST API (Flows)   [ Timeout  ] => Health Check did not finish within 5000 ms

=> Oh no, something is wrong

> ```

Reviewer: ```suggestion
[source, karaf]
----
admin@opennms> health:check
Verifying the health of the container

Verifying installed bundles                    [ Success  ]
Connecting to ElasticSearch ReST API (Flows)   [ Timeout  ] => Health Check did not finish within 5000 ms

=> Oh no, something is wrong
----

```

> OpenNMS deployments install the `opennms-flows` feature by default, but may not point to the Elasticsearch server.

Reviewer: ```suggestion
OpenNMS deployments install the `opennms-flows` feature set by default, but may not point to the Elasticsearch server.
```

> To fix this issue, you can either uninstall the feature (using `feature:uninstall opennms-flows`), or point the feature to an Elasticsearch instance:

Reviewer: ```suggestion
To fix this issue, you can either uninstall the feature (using the `feature:uninstall opennms-flows` command), or point the feature to an Elasticsearch instance:
```

```
config:edit org.opennms.features.flows.persistence.elastic
config:property-set elasticUrl http://elastic:9200
config:update

> ```

Reviewer: ```suggestion
[source, karaf]
----
config:edit org.opennms.features.flows.persistence.elastic
config:property-set elasticUrl http://elastic:9200
config:update
----

```

> TIP: Replace the sample URL with one that points to your Elasticsearch instance.

Reviewer: ```suggestion
NOTE: Replace the sample URL (`http://elastic:9200`) with one that points to your Elasticsearch instance.
```

> Run the health check again to confirm that `everything is awesome`.

Reviewer: ```suggestion
Run the `health:check` command again and, if the "Everything is awesome" message is displayed, continue to the next tasks.
```

=== Enable alarm history storage for Elasticsearch

> When using ALEC, we strongly recommend that you leverage our integration for maintaining alarm history in Elasticsearch.

Reviewer: ```suggestion
We strongly recommended that, when using ALEC, you leverage the Elasticsearch integration.
This integration allows OpenNMS to maintain historical alarm information in Elasticsearch.
```

ALEC can use this data for analysis and to help train the correlation engine.

> .From the OpenNMS Karaf shell, run the following, replacing the sample URL with one that points to your Elasticsearch instance:

Reviewer: ```suggestion
To set up the integration, run the following commands in an OpenNMS Karaf shell:

.Point your OpenNMS instance to an Elasticsearch instance:
```

```
config:edit org.opennms.features.alarms.history.elastic
config:property-set elasticUrl http://elastic:9200
config:update

> ```

Reviewer: ```suggestion
[source, karaf]
----
config:edit org.opennms.features.alarms.history.elastic
config:property-set elasticUrl http://elastic:9200
config:update
----

IMPORTANT: Replace the sample URL (`http://elastic:9200`) with one that points to your Elasticsearch instance.
```

.Use the following to install the feature:

```
feature:install opennms-alarm-history-elastic

> ```

Reviewer: ```suggestion
.Install the feature:
[source, karaf]
feature:install opennms-alarm-history-elastic
```

.Run the `health:check`

> If everything is configured properly, you should receive an `everything is awesome` message.

Reviewer: ```suggestion
After the feature is installed, run the `health:check` command.
If everything is configured properly, an "Everything is awesome" message is displayed.
```

> === Enable Syslogd

Reviewer: ```suggestion
=== Enable syslogd
```

> We recommend that you enable Syslogd to take advantage of the Syslog patterns and event definitions that ALEC provides:

Reviewer: ```suggestion
We recommend that you enable syslogd to take advantage of the syslog patterns and event definitions that ALEC provides.
```

> .Edit `$OPENNMS_HOME/etc/service-configuration.xml`
> and change the following:
> 
> ```
> <service enabled="false">
>     <name>OpenNMS:Name=Syslogd</name>
> ```
> 
> to:
> 
> ```
> <service>
>     <name>OpenNMS:Name=Syslogd</name>
> ```

Reviewer: ```suggestion
First, open `$OPENNMS_HOME/etc/service-configuration.xml` in a text editor and change the following code:

[source]
----
<service enabled="false">
    <name>OpenNMS:Name=Syslogd</name>
----

to:

[source]
----
<service>
    <name>OpenNMS:Name=Syslogd</name>
----
```

This change will come into effect the next time OpenNMS restarts.

> Do not restart OpenNMS at this point, as you will restart it as part of the ALEC installation.

Reviewer: ```suggestion
Do not restart OpenNMS at this point; you will restart it as part of the ALEC installation procedure.
```

> == Set up ALEC/OpenNMS package repositories

Reviewer: ```suggestion
== Set up package repositories
```
